server/storage/files/S3Storage.ts

```diff
@@ -194,28 +194,29 @@ export default class S3Storage extends BaseStorage {
           resolve({ path: tmpFile, cleanup: () => fs.rm(tmpFile) })
         );
 
-        const stream = this.getFileStream(key);
-        if (!stream) {
-          return reject(new Error("No stream available"));
-        }
-
-        stream
-          .on("error", (err) => {
-            dest.end();
-            reject(err);
-          })
-          .pipe(dest);
+        void this.getFileStream(key).then((stream) => {
+          if (!stream) {
+            return reject(new Error("No stream available"));
+          }
+
+          stream
+            .on("error", (err) => {
+              dest.end();
+              reject(err);
+            })
+            .pipe(dest);
+        });
       });
     });
   }
 
-  public getFileStream(key: string): NodeJS.ReadableStream | null {
+  public getFileStream(key: string): Promise<NodeJS.ReadableStream | null> {
     invariant(
       env.AWS_S3_UPLOAD_BUCKET_NAME,
       "AWS_S3_UPLOAD_BUCKET_NAME is required"
     );
 
-    this.client
+    return this.client
```

Comment:
This seemed to be a bug – nothing returned here, but with this change I think it's ready to merge